lib/common/benchmarkModelGenerator.js

```diff
@@ -144,18 +144,16 @@ class BenchmarkModelGenerator {
         let propertyI = 0;
 
         while (!oversized) {
-            const propertiesWithNewAddition = [
-                ...properties,
-                this.generateProperty({
-                    modelI,
-                    declarationI,
-                    propertyI,
-                })
-            ];
+            const generatedProperty = this.generateProperty({
+                modelI,
+                declarationI,
+                propertyI,
+            });
 
-            if (this.jsonSize(propertiesWithNewAddition) <= propertiesSizeBudget) {
+            propertiesSizeBudget = propertiesSizeBudget - this.jsonSize(generatedProperty);
+            if (propertiesSizeBudget >= 0) {
                 propertyI++;
-                properties = propertiesWithNewAddition;
+                properties = [ ...properties, generatedProperty ];
```

Comment:
Given that you're reassigning, why not just `.push`? That way you save duplicating the heap space?

Reply:
Ah, I've not thought about that optimisation before, thanks! Addressed ✅